Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -47,6 +47,32 @@ public struct GraphQLHTTPResponseError: Error, LocalizedError {
   }
 }
 
+public struct GraphQLHTTPRequestError: Error, LocalizedError {
+  public enum ErrorKind {
```

Comment:
You should be able to just make `GraphQLHTTPRequestError` an enum instead of needing a struct wrapper around the embedded enum. I can see we've got this set up in another place similarly but I'm not sure why - @martijnwalraven any thoughts?

Reply:
I did it in that way to confirm with the project's pattern. I didn't want to change it, although I know it's not necessary to put inside a struct wrapper.

Reply:
(samesies - this can stay for this PR)